File: generate_document/generate_document.py
def generate_document(characters, document):
  characters_hash_map = {}
  for character in characters:
    if character in characters_hash_map:
      characters_hash_map[character] += 1
    else:
      characters_hash_map[character] = 1
  for character in document:
    if character not in characters_hash_map or characters_hash_map[character] == 0:
      return False
    characters_hash_map[character] -= 1
  return True

print(generate_document("abc", "cab"))

File: generate_document/generate_document.java
import java.util.HashMap;

public class GenerateDocument {
    public static boolean generateDocument(String characters, String document) {
        HashMap<Character, Integer> charCount = new HashMap<>();
        for (char c : characters.toCharArray()) {
            charCount.put(c, charCount.getOrDefault(c, 0) + 1);
        }

        for (char c : document.toCharArray()) {
            if (!charCount.containsKey(c) || charCount.get(c) == 0) {
                return false;
            }
            charCount.put(c, charCount.get(c) - 1);
        }

        return true;
    }

    public static void main(String[] args) {
        System.out.println(generateDocument("abc", "cab"));
    }
}
